ironfish/src/wallet/scanner/walletScanner.ts

```diff
@@ -93,15 +108,27 @@ export class WalletScanner {
         `Scan starting from block ${scan.start.sequence} to ${scan.start.sequence}`,
       )
 
+      decryptor.start()
+
       void (async () => {
         let hashChanged = true
+
         while (hashChanged) {
-          const head = await this.wallet.getEarliestHead()
-          chainProcessor.hash = head?.hash ?? null
+          if (this.haveWalletAccountsChanged()) {
+            // Accounts have changed in the wallet. Wait for all pending
+            // decrypt requests to be completed, then update the head of the
+            // chain processor.
+            await decryptor.flush()
+            await this.refreshScanningAccounts()
+            const head = this.getEarliestHead()
+            chainProcessor.hash = head?.hash ?? null
```

Comment:
Since this is a long running operation should we check the abort controller here in the scan state?